From 0f49406a5049bd0a3bbc7704bb58efc2b4117424 Mon Sep 17 00:00:00 2001
From: Keerthana K <keerthanak@vmware.com>
Date: Wed, 3 Feb 2021 13:33:34 +0000
Subject: [PATCH] Initialize jitterentropy before ecdh

ECDH uses jitterentropy for seed, so it must initialized
before ecdh for self tests to pass.

---
 crypto/jitterentropy-kcapi.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/crypto/jitterentropy-kcapi.c b/crypto/jitterentropy-kcapi.c
index 68f62be..48d475e 100644
--- a/crypto/jitterentropy-kcapi.c
+++ b/crypto/jitterentropy-kcapi.c
@@ -214,7 +214,8 @@ static void __exit jent_mod_exit(void)
 	crypto_unregister_rng(&jent_alg);
 }
 
-module_init(jent_mod_init);
+/* Must be initialized before tcrypt */
+subsys_initcall(jent_mod_init);
 module_exit(jent_mod_exit);
 
 MODULE_LICENSE("Dual BSD/GPL");
-- 
2.28.0

